source/server/http/admin.h

```diff
@@ -322,6 +322,27 @@ class AdminImpl : public Admin,
                                   AdminStream&);
   bool isFormUrlEncoded(const Http::HeaderEntry* content_type) const;
 
+  class AdminListenSocketFactory : public Network::ListenSocketFactory {
+  public:
+    AdminListenSocketFactory(Network::SocketSharedPtr socket) : socket_(socket) {}
+
+    // Network::ListenSocketFactory
+    Network::Address::SocketType socketType() const override { return socket_->socketType(); }
+
+    const Network::Address::InstanceConstSharedPtr& localAddress() const override {
+      return socket_->localAddress();
+    }
+
+    Network::SocketSharedPtr createListenSocket() override { return socket_; }
```

Comment:
Given this should only be called once, can you either just create the listen socket inside here and return it without a member variable, or set it to null after this call and then assert it's non-null on function entry?

Reply:
Sure sounds good.